### ISSUE TITLE
Since #648 `has_entry` no longer matches by exact value

### ISSUE THREAD

**Issue by ElvinEfendi** [freerange/mocha#654]
Since https://github.com/freerange/mocha/pull/648, `has_entry`'s behaviour changed:

```ruby
require "bundler/setup"
require "minitest/autorun"
require "mocha/minitest"
require "mocha/mock"

class FooTest < Minitest::Test
  def setup
    @mock = mock("foo")
  end

  def test_mocha_mocking_kw_args_with_has_entry
    @mock.expects(:bar).with(has_entry(:a, k: "v"))

    @mock.bar(a: { k: "v", extra: "extra" })
  end
end
```

In the versions before the linked PR the test above would fail because `{ k: "v", extra: "extra" }` is different than the expected `{k: "v"}`. But after that PR the test above succeeds.

**Comment by floehopper** [comment]
@ElvinEfendi Thanks for reporting this - sorry for the slow response - I've been away on leave. I'll take a look as soon as I can.

**Comment by floehopper** [comment]
@ElvinEfendi

I've had an initial look at this and I agree that it is an accidental regression caused by #648.

Many thanks for supplying a test that reproduces the problem.

I've attempted a quick fix in #655.

If you get a chance, could you try out the code in the [`fix-has-entry-with-keyword-args-regression` branch](https://github.com/freerange/mocha/tree/fix-has-entry-with-keyword-args-regression) and see if it fixes your problem...?

**Comment by floehopper** [comment]
@ElvinEfendi I've just released the fix in [v2.4.1](https://rubygems.org/gems/mocha/versions/2.4.1). I hope it solves your problem!
